File: inf-rozsz/2025-maj/src/main.rs
#![allow(missing_docs)]
#![allow(clippy::unwrap_used)]

use std::{
    fs::File,
    io::{BufRead, BufReader},
};

fn is_palindrome(input: &str) -> bool {
    input == input.chars().rev().collect::<String>()
}

fn check_line(vec: &[char], i: usize, character: char) -> bool {
    let prev_char = vec.get(i - 1).unwrap();
    let curr_char = vec.get(i).unwrap();
    let next_char = vec.get(i + 1).unwrap();

    *prev_char == character && *curr_char == character && *next_char == character
}

fn main() {
    let file = File::open("./symbole.txt").unwrap();
    let reader = BufReader::new(file);

    let mut palindrome_count = 0;

    let mut vec: Vec<Vec<char>> = Vec::new();

    for line in reader.lines().map_while(Result::ok) {
        let line = line.trim();

        if is_palindrome(line) {
            println!("{line}");
            palindrome_count += 1;
        }

        vec.push(line.chars().collect());
    }

    for (i, window) in vec.windows(3).enumerate() {
        let prev_line = &window[0];
        let line = &window[1];
        let next_line = &window[2];

        for j in 1..line.len() - 1 {
            let character = line[j];
            let ch1 = check_line(prev_line, j, character);
            let ch2 = check_line(line, j, character);
            let ch3 = check_line(next_line, j, character);

            if ch1 && ch2 && ch3 {
                println!("({},{})", i + 2, j + 1);
            }
        }
    }

    let mut total = 0;
    let mut biggest = 0;

    for line in vec {
        let value = line
            .iter()
            .map(|char| match char {
                'o' => '0',
                '+' => '1',
                '*' => '2',
                _ => unreachable!(),
            })
            .collect::<String>();

        let mut result = 0;
        for c in value.chars() {
            result = result * 3 + c.to_digit(3).unwrap();
        }

        total += result;
        if result > biggest {
            biggest = result;
        }
    }

    println!("Palindrome count: {palindrome_count}");
    println!("Biggest: {biggest}");
    println!("Total: {total}");
}

File: inf-rozsz/2025-maj/src/main.py
def is_palindrome(input: str):
    return input == input[::-1]

def check_line(line: list, i: int, character: str):
    prev_char = line[i-1]
    curr_char = line[i]
    next_char = line[i+1]
    return prev_char == curr_char and curr_char == next_char and curr_char == character

palindrome_count = 0
total = 0
biggest = 0

vec = [[]]

with open("./symbole.txt") as f:
    for i, line in enumerate(f):
        line = line.strip()

        if is_palindrome(line):
            print(line)
            palindrome_count+=1

        for char in line:
            if len(vec) <= i:
                vec.append([char])
            else:
                vec[i].append(char)

    for i in range(1, len(vec)-1):
        prev_line = vec[i-1]
        line = vec[i]
        next_line = vec[i+1]
        for j in range(1, len(line)-1):
            character = line[j]
            ch1 = check_line(prev_line, j, character)
            ch2 = check_line(line, j, character)
            ch3 = check_line(next_line, j, character)

            if ch1 and ch2 and ch3:
                print(f"({i+1},{j+1})")

    for i in range(len(vec)):
        value = ""

        for j in range(len(vec[i])):
            character = vec[i][j]

            if character == 'o':
                value += "0"
            elif character == '+':
                value += "1"
            elif character == '*':
                value += "2"

        result = 0
        for c in value:
            result = result * 3 + int(c)

        total += result
        if result > biggest:
            biggest = result


print(f"Palindrome count: {palindrome_count}")
print(f"Biggest: {biggest}")
print(f"Total: {total}")
